presto-memory/src/main/java/io/prestosql/plugin/memory/MemoryConnector.java

```diff
@@ -44,6 +44,13 @@ public MemoryConnector(
         this.pageSinkProvider = pageSinkProvider;
     }
 
+    @Override
+    public boolean isSingleStatementWritesOnly()
+    {
+        // TODO (https://github.com/prestosql/presto/issues/5467) fix
```

Comment:
Returning `true` here should fix that issue, since this will disallow writes within a transaction, so we can remove this comment and close the issue after this is merged.